src/views/engine-app/flow_p0002_diagrams_vw.sql

```diff
@@ -1,21 +1,22 @@
 
 create or replace view flow_p0002_diagrams_vw
 as
-  select 
-    dgrm_id
-    , dgrm_name
-    , dgrm_version
-    , dgrm_status
-    , dgrm_category
-    , dgrm_last_update
-    , '<button type="button" title="View" aria-label="View" class="clickable-action t-Button t-Button--noLabel t-Button--icon" data-dgrm="' || dgrm_id || '" data-action="dgrm_view"><span aria-hidden="true" class="t-Icon fa fa-eye"></span></button>'||
-      case dgrm_status when 'draft' then '<button type="button" title="Edit" aria-label="Edit" class="clickable-action t-Button t-Button--noLabel t-Button--icon" data-dgrm="' || dgrm_id || '" data-action="dgrm_edit"><span aria-hidden="true" class="t-Icon fa fa-pencil"></span></button>' end||
-      '<button type="button" title="New version" aria-label="New version" class="clickable-action t-Button t-Button--noLabel t-Button--icon" data-dgrm="' || dgrm_id || '" data-action="dgrm_new_version"><span aria-hidden="true" class="t-Icon fa fa-plus"></span></button>'||
-      case dgrm_status when 'draft' then '<button type="button" title="Release" aria-label="Release" class="clickable-action t-Button t-Button--noLabel t-Button--icon" data-dgrm="' || dgrm_id || '" data-action="dgrm_release"><span aria-hidden="true" class="t-Icon fa fa-check"></span></button>' end||
-      case dgrm_status when 'released' then '<button type="button" title="Deprecate" aria-label="Deprecate" class="clickable-action t-Button t-Button--noLabel t-Button--icon" data-dgrm="' || dgrm_id || '" data-action="dgrm_deprecate"><span aria-hidden="true" class="t-Icon fa fa-ban"></span></button>' end||
-      case dgrm_status when 'deprecated' then '<button type="button" title="Archive" aria-label="Archive" class="clickable-action t-Button t-Button--noLabel t-Button--icon" data-dgrm="' || dgrm_id || '" data-action="dgrm_archive"><span aria-hidden="true" class="t-Icon fa fa-archive"></span></button>' end
+  select distinct
```

Comment:
Why do we need distinct here?

Reply:
Adding left join with flow_instances_vw in order to display total instance associated to the diagram :
count(i.prcs_id) over (partition by d.dgrm_id) as instances
So distinct is used to display one line per diagram

Reply:
If there is a better way, please let me know.

Reply:
I would use a scalar sub select.
`coalesce( (select count(*) from flow_instances_vw i where i.dgrm_id = d.dgrm_id), 0 ) as instances`
and remove the left join

Reply:
Are you doing the change during the merge or do you want a new pulled request?

Reply:
Do you want to correct flow_p0002_instances_counter_vw too?

Reply:
> Are you doing the change during the merge or do you want a new pulled request?

You can update the code on your side and push again.
This will get picked up as an updated pull request in here.



> Do you want to correct flow_p0002_instances_counter_vw too?

Depends on what should be achieved here.
If we want a row for all known diagrams then the current is probably fine.
If only diagrams with at least 1 instance should be shown we can eliminate the join to flow_diagrams and only query flow_instance_vw with group by on flow_instances_vw.dgrm_id

Reply:
Ok I will pushed again.
For flow_p0002_instances_counter_vw, the aim is to provide a counter on the diagram page to show for the selected diagram, number of:
- created instance
- running instance
- completed instance
![image](https://user-images.githubusercontent.com/42840524/106487891-6caaf980-64b3-11eb-984c-145c5b42477a.png)

Reply:
View updated